docs/articles/reference/upgrading-to-v16.md

```diff
@@ -0,0 +1,258 @@
+
+### Windowing
+
+Most of the operators from `v0.15` are the same, but the `start_at` parameter of windowing functions has been changed to `align_to`. The `start_at` parameter incorrectly implied that there were no potential windows before that time. What determines if an item is late for a window is not the windowing definition, but the watermark of the clock.
+
+`SlidingWindow` and `TumblingWindow` now require the align_to parameter. Previously, this was filled with the timestamp of the start of the Dataflow, but must now be specified. Specifying this parameter ensures that windows are consistent across Dataflow restarts, so make sure that you don't change this parameter between executions.
+
+
+So the old `TumblingWindow` definition:
+
+```python doctest:SKIP
+clock_config = SystemClockConfig()
+window_config = TumblingWindowConfig(length=timedelta(seconds=5))
+```
+
+becomes:
+
+```python doctest:SKIP
+clock_config = SystemClockConfig()
+window_config = TumblingWindow(
+    length=timedelta(seconds=5), align_to=datetime(2023, 1, 1, tzinfo=timezone.utc)
+)
+```
+
+### Output and execution
+
+Similarly to the `input`, the output configuration is now stateful. `capture` has been renamed to `output`, and now takes a name, as all stateful operators do.
+
+So we move from this:
+
+```python doctest:SKIP
+flow.capture(StdOutputConfig())
+```
+
+To this:
+
+```python doctest:SKIP
+flow.output("out", StdOutput())
+```
+
+### Complete code
+The complete code for the new simple example now looks like this:
+
+```python doctest:SKIP
+import operator
+import re
+
+from datetime import timedelta, datetime, timezone
+
+from bytewax.dataflow import Dataflow
+from bytewax.connectors.files import FileInput
+from bytewax.connectors.stdio import StdOutput
+from bytewax.window import SystemClockConfig, TumblingWindow
+
+
+def lower(line):
+    return line.lower()
+
+
+def tokenize(line):
+    return re.findall(r'[^\s!,.?":;0-9]+', line)
+
+
+def initial_count(word):
+    return word, 1
+
+
+def add(count1, count2):
+    return count1 + count2
+
+
+clock_config = SystemClockConfig()
+window_config = TumblingWindow(
+    length=timedelta(seconds=5), align_to=datetime(2023, 1, 1, tzinfo=timezone.utc)
+)
+
+flow = Dataflow()
+flow.input("inp", FileInput("examples/sample_data/wordcount.txt"))
+flow.map(lower)
+flow.flat_map(tokenize)
+flow.map(initial_count)
+flow.reduce_window("sum", clock_config, window_config, add)
+flow.output("out", StdOutput())
+```
+
+Running our completed Dataflow looks like this, as we've named our Dataflow variable `flow` in a file named `dataflow`:
+
+``` bash
+python -m bytewax.run dataflow:flow
+('whether', 1)
+("'tis", 1)
+('of', 2)
+('opposing', 1)
+```
+
+We can even run our sample Dataflow on multiple workers to process the file in parallel:
+
+In one terminal, run:
+
+``` bash
+> python -m bytewax.run dataflow:flow -i0 -a "localhost:2101;localhost:2102"
```

Comment:
Is this better demonstrated via the "spawn cluster" `-p` flag, rather than starting two workers by hand?